rust/tests/ffi.rs

```diff
@@ -0,0 +1,184 @@
+#![allow(clippy::undocumented_unsafe_blocks, unused_crate_dependencies)]
+use std::{ffi, ptr};
+
+use driver::{
+    get_tx_context_zeroed,
+    host_interface::{self, null_ptr_host_interface},
+    TX_CONTEXT_ZEROED, ZERO,
+};
+use evmc_vm::{Revision, StatusCode, StepStatusCode};
+use evmrs::{MockExecutionContextTrait, MockExecutionMessage, Opcode};
+
+#[test]
+fn run_with_mock_context() {
+    let host = host_interface::mocked_host_interface();
+    let mut context = MockExecutionContextTrait::new();
+    context
+        .expect_get_tx_context()
+        .times(1)
+        .return_const(TX_CONTEXT_ZEROED);
+    let revision = Revision::EVMC_CANCUN;
+    let message = driver::to_evmc_message(&MockExecutionMessage::default());
+    let code = &[Opcode::Push0 as u8];
+    let result = unsafe {
+        driver::run(
+            &host,
+            &mut context as *mut MockExecutionContextTrait as *mut ffi::c_void,
+            revision,
+            &message,
+            code,
+        )
+    };
+    assert_eq!(result.status_code, StatusCode::EVMC_SUCCESS);
+}
+
+#[test]
+fn run_with_only_tx_context() {
+    let mut host = null_ptr_host_interface();
+    host.get_tx_context = Some(get_tx_context_zeroed);
+    let context = ptr::null_mut();
+    let revision = Revision::EVMC_CANCUN;
+    let message = driver::to_evmc_message(&MockExecutionMessage::default());
+    let code = &[Opcode::Push0 as u8];
+    let result = unsafe { driver::run(&host, context, revision, &message, code) };
+    assert_eq!(result.status_code, StatusCode::EVMC_SUCCESS);
+}
+
+#[test]
+fn run_no_code() {
```

Comment:
overlaps with run_steppable_without_values

Reply:
No because run_no_code goes though the non-steppable interface and run_steppable_without_values goes through the steppable interface.